### PR TITLE
Add contribution guidance document for Java Azure function

### PR DESCRIPTION
**Pull request by helayoty**
In this PR:
- Add CONTRIBUTOR.md for developers to setup a debugging environment for the Azure function java worker.
- A screenshot for configuring the debugging environment for azure function worker in IntelliJ

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+- For each PR request/merge, a continuous integration pipeline will run internally that performs linting and running unit tests on your PR/merge.
+- A GitHub Action will also perform CI tasks against your PR/merge. This is designed to provide more control to the contributor.
+
+## Getting help
+
+ - Leave comments on your PR and @username for attention
+
+### Requesting a release
+- If you need a release into maven central, request it by raising an issue and tagging @anthonychu or @davidmrdavid
```
Comment (TsuyoshiUshio) [inline CONTRIBUTING.md:115]: It is ok for @TsuyoshiUshio  and @amamounelsayed .

**Review by TsuyoshiUshio — COMMENTED** [review]
I leave one comment. What do you think? @amamounelsayed ?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+- [Azure Functions Java Overview](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-azure-function-azure-cli?tabs=bash%2Cbrowser&pivots=programming-language-java)
+- [Azure Function Java developer guide](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-java?tabs=consumption)
+
+## Pre-requisites
+
+- OS
+    - MacOS, Ubuntu (or) Windows10
+- Language Runtimes
+    - .NET Core 2.0
+    -  Java 1.8
```
Comment (amamounelsayed) [inline CONTRIBUTING.md:24]: I believe JDK will be enough for Java Tools, or move the tools to here.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+- IDE
+   - [IntelliJ](https://www.jetbrains.com/idea/download)
+   - [Java extension in VS Code](https://code.visualstudio.com/docs/languages/java)
+- Java Tools 
+    - [Maven](https://maven.apache.org/install.html)
+    - [JDK (openJDK)  8 and/or 11](https://openjdk.java.net/install/)
```
Comment (amamounelsayed) [inline CONTRIBUTING.md:31]: We use Zulu OpenJDK. https://docs.microsoft.com/en-us/azure/developer/java/fundamentals/java-jdk-long-term-support

**Review by amamounelsayed — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -7,7 +7,7 @@
 
 # Contributing
 
-This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+Please refer to [CONTRIBUTING.md](./CONTRIBUTING.md) for more information.
```
Comment (priyaananthasankar) [inline README.md:10]: Better to redirect to OpenJDK or provide the link to the runtime or jdk download.

Comment (priyaananthasankar) [reply README.md:10]: Also need specific instructions to install JDK in MacOS/Windows/Linux atleast. I used brew

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    | Variable   |  Value    |
+    | :--------: | :------:  |
+    | FUNCTIONS_WORKER_RUNTIME       | java |
+    | languageWorkers:java:arguments | -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005 |
+    | AZURE_FUNCTIONS_ENVIRONMENT    | Development |
+    | AzureWebJobsScriptRoot         | a path to your azure function target folder from step 1,i.e. ~/< your-folder-path >/target/azure-functions/<azure-function-name-####> |
+    
+    >> Note: In macOS, you might need to add JAVA_HOME as a debugging environment variable. 
+
+6. Navigate to WorkerProcess class and add breakpoint at `_process.Start()`. Run the host in a debugging mood
```
Comment (priyaananthasankar) [inline CONTRIBUTING.md:91]: "mode" instead of mood (typo)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+8. In case you need to test a new version from Java worker code, you'll need to:
+    a. Go to target folder and rename `azure-functions-java-worker-1.8.0.jar` to `azure-functions-java-worker.jar`
+    b. Copy this jar file and paste it in the debug folder in the host code base(<Azure_Function_Host_Path>/azure-functions-host/src/WebJobs.Script.WebHost/bin/Debug/netcoreapp3.1/workers/java)
+    
+9. Add new remote debugging configuration as show below
+
+![](tools/.images/worker-debug-configuration.png)
+
+10. Navigate to `JavaWorkerManager.java` and add a breakpoint in the constructor. Run the worker in the debugging mode. This should internally start the Azure Function using `func host start` command.
```
Comment (priyaananthasankar) [inline CONTRIBUTING.md:105]: I cannot find JavaWorkerManager anywhere. Did you mean JavaWorkerClient?

**Comment by helayoty** [comment]
@amamounelsayed  & @TsuyoshiUshio  The PR is ready for final review.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+The general flow for making a change to the library is:
+
+1. 🍴 Fork the [worker](https://github.com/helayoty/azure-functions-java-worker) repo (add the fork via `git remote add me <clone url here>`
+2. 🌳 Create a branch for your change (generally branch from dev) (`git checkout -b my-change`)
+3. 🛠 Make your change
+4. ✔️ Test your change
+5. ⬆️ Push your changes to your fork (`git push me my-change`)
+6. 💌 Open a PR to the dev branch
+7. 📢 Address feedback and make sure tests pass (yes even if it's an "unrelated" test failure)
```
Comment (TsuyoshiUshio) [inline CONTRIBUTING.md:53]: For the current CI, The forked repo users need to ask us to run the CI pipeline. Then I or Ahmed will run the pipeline.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+1. Add your unit tests under ./src/test folder
+2. Run: `mvn clean package`
+
+## Continuous Integration Guidelines & Conventions
+
+This project uses a combination of Azure DevOps and GitHub Actions for CI/CD.
+
+- For each PR request/merge, a continuous integration pipeline will run internally that performs linting and running unit tests on your PR/merge.
+- A GitHub Action will also perform CI tasks against your PR/merge. This is designed to provide more control to the contributor.
```
Comment (TsuyoshiUshio) [inline CONTRIBUTING.md:184]: Do we have GitHub Action on this repo?

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+### Running unit tests
+
+1. Add your unit tests under ./src/test folder
+2. Run: `mvn clean package`
+
+## Continuous Integration Guidelines & Conventions
+
+This project uses a combination of Azure DevOps and GitHub Actions for CI/CD.
+
+- For each PR request/merge, a continuous integration pipeline will run internally that performs linting and running unit tests on your PR/merge.
```
Comment (TsuyoshiUshio) [inline CONTRIBUTING.md:183]: Same as above. Non-Forked user's PR will run automatically, forked repo user need to ask us to run the pipeline.

**Review by TsuyoshiUshio — APPROVED** [review]
Thank you for your great contribution. 
I just leave some comments. However, I approve it since I don't like long-time-open-pr. :)

File: CONTRIBUTING.md (hunk 8 above)
Comment (helayoty) [reply CONTRIBUTING.md]: Removed!

**Review by amamounelsayed — APPROVED** [review]

**Pull request merged**